Modules/Compatibility/Deprecated/include/itkVectorResampleImageFilter.h

```diff
@@ -233,14 +233,14 @@ class ITK_TEMPLATE_EXPORT VectorResampleImageFilter : public ImageToImageFilter<
 
 
 private:
-  SizeType                m_Size;
-  TransformPointerType    m_Transform;
-  InterpolatorPointerType m_Interpolator;
-  PixelType               m_DefaultPixelValue;
-  SpacingType             m_OutputSpacing;
-  OriginPointType         m_OutputOrigin;
-  DirectionType           m_OutputDirection;
-  IndexType               m_OutputStartIndex;
+  SizeType                m_Size{};
```

Comment:
Will these, the lists, the vectors or the arrays make unhappy Sean's AppleClang builds? I would dislike seeing new warnings because we are missing braces. I know that at this point the size of the size might not be known, but I ignore how AppleClang thinks.

Reply:
> Maybe I'm wrong and I have not had a look at the class of the m_Exporter pointer, but as you say it is a smart pointer, and these initialize themselves to the null pointer. Why is then this necessary? If it is, then it needs to be explained in the SW Guide; if it is an effect of the regex, then smart pointer cases should be revised/reverted IMO.

@jhlegarreta Indeed, for an `itk::SmartPointer`, the addition of an `{}` initializer has no effect at run-time. It's really a matter of style and consistency to add such an initializer to each and every data member. In my opinion, it will improve both robustness and code readability if we just add those initializers to _all_ data members, without having to distinguish between different types of data members.

Yes, I very much agree with you that this should be explained in the SW Guide.

Just to be sure, I double-checked for you: adding an empty pair of braces `{}` as initializer does _not_ introduce a clang warning (like "suggest braces around initialization of subobject [-Wmissing-braces]") 👍 Having tried the following code at https://godbolt.org/z/ofzxKzfan

```
struct Size
{
  int m_Data[3];
};

class MyClass
{
public:
  Size m_MissingBraces{1, 2}; // <== An old Clang warning here.
  Size m_ValueInitialized{};  // <== No Clang warning here  :-)
};

``` 
So that shouldn't be a problem 😃

Reply:
OK. So I agree, the consistency and readability arguments are powerful in favor of this. Thanks Niels.